File: 03. Programming - C#/TestPreparation4Example/Shape.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TestPreparation4Example
{
    abstract class Shape
    {
        private string name;
        public string Name
        {
            get
            {
                return name;
            }
            set
            {
                name = value;
            }
        }

        private int numberOfSides;
        public int NumberOfSides
        {
            get
            {
                return numberOfSides;
            }
            set
            {
                numberOfSides = value;
            }
        }

        private string type;
        public string Type
        {
            get
            {
                return type;
            }
            set
            {
                type = value;
            }
        }

        public abstract double calculateArea();
        public abstract double calculatePerimeter();

        public Shape(string name, int numberOfSides,string type)
        {
            this.name = name;
            this.numberOfSides = numberOfSides;
            this.type = type;
        }
    }
}

File: 02. Programming - Java/Lesson7/src/uni/fmi/bachelors/Shape.java
package uni.fmi.bachelors;

public abstract class Shape {

	private String name;
	private int numberOfSides;
	private String type;

	public Shape(String type, int numberOfSides) {
		this.type = type;
		this.numberOfSides = numberOfSides;
	}

	public abstract double calculatePerimeter();

	public abstract double calculateArea();

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public int getNumberOfSides() {
		return numberOfSides;
	}

	public void setNumberOfSides(int numberOfSides) {
		this.numberOfSides = numberOfSides;
	}

	public String getType() {
		return type;
	}

	public void setType(String type) {
		this.type = type;
	}

}
